gen/list.go

```diff
@@ -147,3 +148,49 @@ func (l *listGenerator) Reader(g Generator, spec *compile.ListSpec) (string, err
 
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
+
+// Equals generates a function to compare lists of the given type
+//
+// func $name(lhs, rhs $listType) bool {
+//      ...
+// }
+//
+// And returns its name.
+func (l *listGenerator) Equals(g Generator, spec *compile.ListSpec) (string, error) {
+	name := "_" + valueName(spec) + "_Equals"
+	if l.HasEquals(name) {
+		return name, nil
+	}
+
+	err := g.DeclareFromTemplate(
+		`
+            <$listType := typeReference .Spec>
+
+			<$lhs := newVar "lhs">
+			<$rhs := newVar "rhs">
+			<$i := newVar "i">
+			<$lv := newVar "lv">
```

Comment:
inconsistent indentation. make sure your editor is using tabs here.
enable visible whitespace to verify this.